[APM] Adds button group to navigate to "All services"

ogupte opened the pull request:
Closes #142084.

![apm-service-groups-button-group-nav4](https://user-images.githubusercontent.com/1967266/198731047-a8f4370a-28ca-4ece-99d9-1a575cf799ee.gif)

ogupte committed:
[APM] Adds button group to navigate to "All services" (#142084)

elasticmachine commented:
Pinging @elastic/apm-ui (Team:APM)

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/precommit_hook.js --ref HEAD~1..HEAD --fix'

ogupte committed:
Adds button group legend, fixes spacing and translation IDs

bmorelli25 commented on x-pack/plugins/apm/public/components/app/service_groups/service_groups_list/index.tsx:
```diff
@@ -21,11 +22,56 @@ import { ServiceGroupsListItems } from './service_groups_list';
+            'xpack.apm.serviceGroups.tour.serviceGroups.content',
+            {
+              defaultMessage:
+                "Now that you've created a service group, you can go back and view all services here.",
```
I wonder if this tour element is even needed anymore. With a button title of `View all services`, I don't think the tour text adds anything that the button doesn't already communicate.

ogupte committed:
makes the button group part of the page template

ogupte committed:
Merge branch 'main' into apm-142084-service-groups-button-group-nav

ogupte committed:
addresses feedback

ogupte committed:
Merge branch 'main' into apm-142084-service-groups-button-group-nav

sorenlouv reviewed (approved):
lgtm

sqren commented on x-pack/plugins/apm/public/components/app/service_groups/service_groups_button_group.tsx:
```diff
@@ -0,0 +1,63 @@
+    <>
+      <EuiButtonGroup
+        color="primary"
+        type="multi"
```
Why "multi"? Users shouldn't be able to select both options, should they?

ogupte committed:
Merge branch 'main' into apm-142084-service-groups-button-group-nav

ogupte committed:
addressed feedback:
- Use 'All services' as the default landing page
    - swap order of button group ->  All services, Service groups
    - move tour from the create button to service group half of the button group
- have service groups always enabled but marked Beta
    - removed some branching logic in routes / page templates
    - add 'Send feedback' link
- fixed service count in cards

ogupte committed:
Merge branch 'main' into apm-142084-service-groups-button-group-nav

ogupte committed:
fixes tests and deep links

ogupte committed:
change button group to single type

ogupte committed:
fixes type error

ogupte committed:
Merge branch 'main' into apm-142084-service-groups-button-group-nav

gbamparop commented on x-pack/plugins/apm/public/components/app/service_groups/service_group_save/create_button.tsx:
```diff
@@ -7,42 +7,42 @@
-            'Group services together to build curated inventory views that remove noise and simplify investigations across services. Groups are Kibana space-specific and available for any users with appropriate access.',
-        }
-      )}
+    // <ServiceGroupsTour
```
@ogupte could this be removed instead?

ogupte replied:
whoops, i'll remove it while working on alerts integration
